spec/system/composer/morphed_preview_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+describe "Morphed Composer Preview", type: :system do
+  fab!(:user) { Fabricate(:user, username: "bob", refresh_auto_groups: true) }
+  let(:composer) { PageObjects::Components::Composer.new }
+
+  before do
+    SiteSetting.enable_diffhtml_preview = true
+    sign_in user
+    visit("/new-topic")
+  end
+
+  it "correctly morphs code blocks" do
+    composer.fill_content <<~MD
+      ```js
+      const = {
+        id: t.name,
+        text: t.name,
+        name: t.name,
+      ```
+    MD
+
+    within(composer.preview) { expect(find("code.lang-js")).to have_text("const = {") }
+
+    composer.move_cursor_after("const")
+    composer.type_content("ant")
+
+    within(composer.preview) { expect(find("code.lang-js")).to have_text("constant = {") }
```

Comment:
Does highlightjs work correctly now that we moved the decorating to the 'wrong' elements? (highlightjs is loaded async, so it won't necessarily happen when `decorateCooked()` is triggered)

Reply:
It does but it’s hard to test correctly. I felt this test wasn’t about testing highlightjs though so I switch the selector back to the markdown one.